File: backend/app/schemas/notification.py
"""通知相关 Pydantic 模型。"""
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class NotificationOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    type: str
    actor_id: int | None = None
    actor_nickname: str = ""
    actor_avatar: str = ""
    community_id: int | None = None
    community_name: str = ""
    ref_id: int | None = None
    title: str = ""
    summary: str = ""
    is_read: bool = False
    read_at: datetime | None = None
    created_at: datetime | None = None


class NotifySettingsUpdate(BaseModel):
    """通知开关（键与 users.notify_settings 一致，见 详细开发方案.md 文档⑬）。"""

    model_config = {"extra": "forbid"}  # 未知开关键直接 400，不静默忽略

    mention: bool | None = None
    like: bool | None = None
    comment: bool | None = None
    follow: bool | None = None
    system: bool | None = None
    review: bool | None = None
    report: bool | None = None

File: frontend/src/api/notification.ts
/** 通知相关 API（阶段 5）。 */
import { request, type Page } from '@/api/http'

export interface NotificationItem {
  id: number
  type: 'mention' | 'like' | 'comment' | 'follow' | 'system' | 'review_result' | 'report_feedback'
  actor_id: number | null
  actor_nickname: string
  actor_avatar: string
  community_id: number | null
  community_name: string
  ref_id: number | null
  title: string
  summary: string
  is_read: boolean
  read_at: string | null
  created_at: string | null
}

export interface NotifySettings {
  mention: boolean
  like: boolean
  comment: boolean
  follow: boolean
  system: boolean
  review: boolean
  report: boolean
}

export const notificationApi = {
  list: (page = 1, page_size = 20) =>
    request<Page<NotificationItem>>({ url: '/notifications', params: { page, page_size } }),
  unreadCount: () => request<{ count: number }>({ url: '/notifications/unread-count' }),
  read: (id: number) => request({ url: `/notifications/${id}/read`, method: 'POST' }),
  readAll: () => request<{ marked: number }>({ url: '/notifications/read-all', method: 'POST' }),
  remove: (id: number) => request({ url: `/notifications/${id}`, method: 'DELETE' }),
  getSettings: () => request<NotifySettings>({ url: '/notifications/settings' }),
  updateSettings: (patch: Partial<NotifySettings>) =>
    request<NotifySettings>({ url: '/notifications/settings', method: 'PUT', data: patch }),
}
